pennylane_lightning/core/src/simulators/lightning_tensor/cutn/MPSCutn.hpp

```diff
@@ -0,0 +1,341 @@
+        }
+        return tensorsDataPtr;
+    }
+
+    /**
+     * @brief Set a zero state
+     */
+    void reset() {
+        const std::vector<std::size_t> zeroState(BaseType::getNumQubits(), 0);
+        setBasisState(zeroState);
+    }
+
+    /**
+     * @brief Set basis state
+     * NOTE: This API assumes the bond vector is [1,0,0,......] and current
+     * implementation only works for qubits.
+     * @param basisState Vector representation of a basis state.
+     */
+    void setBasisState(const std::vector<std::size_t> &basisState) {
+        PL_ABORT_IF(BaseType::getNumQubits() != basisState.size(),
+                    "The size of a basis state should be equal to the number "
+                    "of qubits.");
+
+        bool allZeroOrOne = std::all_of(
+            basisState.begin(), basisState.end(),
+            [](std::size_t bitVal) { return bitVal == 0 || bitVal == 1; });
+
+        PL_ABORT_IF(allZeroOrOne == false,
+                    "Please ensure all elements of a basis state should be "
+                    "either 0 or 1.");
+
+        PL_ABORT_IF(this->MPSInitialized_ == MPSStatus::MPSInitSet,
+                    "setBasisState() can be called only once.");
+
+        this->MPSInitialized_ = MPSStatus::MPSInitSet;
+
+        CFP_t value_cu =
+            Pennylane::LightningGPU::Util::complexToCu<ComplexT>({1.0, 0.0});
+
+        for (std::size_t i = 0; i < BaseType::getNumQubits(); i++) {
+            tensors_[i].getDataBuffer().zeroInit();
+            std::size_t target = 0;
+            std::size_t idx = BaseType::getNumQubits() - std::size_t{1} - i;
+
+            // Rightmost site
+            if (i == 0) {
+                target = basisState[idx];
+            } else {
+                target = basisState[idx] == 0 ? 0 : maxBondDim_;
+            }
+
+            PL_CUDA_IS_SUCCESS(
+                cudaMemcpy(&tensors_[i].getDataBuffer().getData()[target],
+                           &value_cu, sizeof(CFP_t), cudaMemcpyHostToDevice));
+        }
+
+        updateQuantumStateMPS_(getSitesExtentsPtr().data(),
+                               getTensorsDataPtr().data());
+    };
+
+    /**
+     * @brief Get the full state vector representation of MPS quantum state.
+     *
+     * NOTE: This API is only for MPS unit tests purpose only, given that the
+     * full statevector system is small. Attempt to apply this method to largr
+     * systems will lead to memory issue.
+     *
+     * @return Full state vector representation of MPS quantum state on host
+     * std::vector<ComplexT>
+     */
+    auto getDataVector() -> std::vector<ComplexT> {
+        PL_ABORT_IF(this->MPSFinalized_ == MPSStatus::MPSFinalizedSet,
+                    "getDataVector() method to return the full state "
+                    "vector can't be called "
+                    "after cutensornetStateFinalizeMPS is called");
+
+        this->MPSFinalized_ = MPSStatus::MPSFinalizedSet;
```

Comment:
```suggestion
        MPSFinalized_ = MPSStatus::MPSFinalizedSet;
```

Same for all other `this` uses here and throughout